api/dispatcher.proto

```diff
@@ -202,4 +202,14 @@ message AssignmentsMessage {
 	// assignment set. It is not used in the first assignments message of
 	// a stream.
 	repeated string remove_tasks = 5;
+
+	// UpdateSecrets is a set of new or updated secrets for this node.
+	// In the first assignments message, it contains all of the secrets
+	// the node needs for itself and its assigned tasks.
+	repeated Secret update_secrets = 6;
```

Comment:
Too late now, but we should have used one level of indirection.

Assignments should be part of the data model - **not** the RPC protocol as it is now.

What I mean by that is we should have had an `Assignment` object type with a `oneof`. The dispatcher protocol therefore (`AssignmentsMessage`) should have been using that instead (rather than embedding a pseudo definition of assignments inline).

With this approach, adding a new type of assignment requires making changes to the dispatcher RPC which is broken.

This is not novel - that's what we're already doing for Raft:
https://github.com/docker/swarmkit/blob/master/api/raft.proto#L119

/cc @stevvooe @aaronlehmann